src/transformers/generation/configuration_utils.py

```diff
@@ -569,9 +570,9 @@ def from_dict(cls, config_dict: Dict[str, Any], **kwargs) -> "GenerationConfig":
         if "_commit_hash" in kwargs and "_commit_hash" in config_dict:
             kwargs["_commit_hash"] = config_dict["_commit_hash"]
 
-        # remove all the arguments that are in the config_dict
-
-        config = cls(**config_dict, **kwargs)
+        # The line below allows model-specific config to be loaded as well through kwargs, with safety checks.
+        # See https://github.com/huggingface/transformers/pull/21269
+        config = cls(**{**config_dict, **kwargs})
```

Comment:
Before: `cls` could receive two keyword arguments with the same name
Now: builds a dictionary where the manually defined attributes take precedence before passing it downstream